app/workers/facilities/dental_service_reload_job.rb

```diff
@@ -21,25 +18,18 @@ def fetch_dental_service_data(file)
       raise DentalServiceError, "Failed to read CSV file: #{file} - caused by: #{e.cause}"
     end
 
-    def update_cache(model, facilities)
+    def update_cache(facilities)
       facilities.each do |k|
         attrs = { station_number: k,
                   local_updated: Time.now.utc.iso8601 }
-        obj = model.find(k)
-        if obj
-          obj.update(attrs)
-        else
-          model.create(attrs)
-        end
+        obj = FacilityDentalService.find_or_build(k)
+        obj.update(attrs)
       end
-
-      logger.info "Updated facility dental service cache for #{facilities.size} facilities"
     end
 
-    def invalidate_removed(model, facility_keys)
-      invalidate = model.keys - facility_keys
-      invalidate.each { |x| model.delete(x) }
-      logger.info "Removed #{invalidate.size} obsolete entries from cache"
+    def invalidate_removed(facility_keys)
+      invalidate = FacilityDentalService.keys - facility_keys
+      invalidate.each { |x| FacilityDentalService.delete(x) }
```

Comment:
Maybe rename `invalidate` to `removed_keys`? (If that's actually what this code means - I can't quite tell from the names.) Or, if it mean invalid keys, then could we name it that? I.e., the plural noun, `invalid_keys`

Reply:
Ah hah, from the deleted debug message (!) I'm inferring that this is doing the opposite of what the function name implied to me: `invalidate_removed(facility_keys)`. I.e., I read that to mean, "Invalidate the removed facility keys". But it's actually, "Remove the invalid facility keys" (?) If that's true, I suggest we rename the function as well.